src/Microsoft.DotNet.Internal.DependencyInjection.Testing/ServiceDescriptorExtension.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.DotNet.Internal.DependencyInjection.Testing;
+public static class ServiceDescriptorExtension
+{
+    public static Type GetImplementationType(this ServiceDescriptor descriptor) =>
```

Comment:
Thoughts on making this `internal`?

Reply:
I'm not sure I can think of any downsides, maybe someone who is using this library won't expect this class to do what it does?
In my opinion, this is quite a useful method, I'm surprised it's not already in there

Reply:
Well, the downsides is that if you start using `Microsoft.DotNet.Internal.DependencyInjection.Testing`, you get a lot of things from it that you shouldn't see. Ideally it's best to keep the API surface small. So if this is only needed here, I'd make it `internal`. I would even consider to just make it a private method in the DI test as a regular method.

It's much better to have this piece of code tied to only one thing living with it. It's impossible to kill dead code that is public static like this and without the knowledge of how it's used and where (if it can be external).

Reply:
Yeah that does make sense, it's like a side thing that a person using this package is not interested in. I made it a private method, I don't think there's a need to make it an extension